### ISSUE TITLE
Question Regarding Potential Fairness Issue with Loading Pre-trained Model

### ISSUE THREAD

**Issue by yyq2215986391** [zhangqf-lab/SPACE#1]
Hello, I have a question about a specific part of the code. Here is the segment that caught my attention:
`
device = train_SPACE(model, train_data, epoch=epoch, lr=lr, patience=patience, GPU=GPU, seed=seed, a=alpha, loss_type=loss_type, outdir=outdir, verbose=verbose)

# Load model
pretrained_dict = torch.load(os.path.join(outdir, 'model.pt'), map_location=device)
model_dict = model.state_dict()
pretrained_dict = {k: v for k, v in pretrained_dict.items() if k in model_dict}
model_dict.update(pretrained_dict)
model.load_state_dict(model_dict)
model = model.eval()
`
Could there be a potential fairness issue with reloading the pre-trained model immediately after training it? My concern is that this approach might not properly separate the training and evaluation phases, possibly leading to inflated performance metrics due to the absence of an independent validation or testing step.
